src/Pact/Types/Runtime.hs

```diff
@@ -242,12 +241,6 @@ newtype Eval e a =
     deriving (Functor,Applicative,Monad,MonadState EvalState,
                      MonadReader (EvalEnv e),MonadThrow,MonadCatch,MonadIO)
 
-instance Semigroup a => Semigroup (Eval e a) where
-  (<>) = liftA2 (<>)
-
-instance Monoid a => Monoid (Eval e a) where
-  mempty = pure mempty
-
```

Comment:
It turns out that this did not play well with laziness, and failed silently. It's a shame, but we can always use `foldr` instead. That said, that means we should watch out in the future, because the monoid laws for `Eval` do not hold

Reply:
Someday, i will have the galaxy brain necessary to debug it and give a full report. Maybe we can do it as an exercise if you have free time, because i'm at a loss regarding how to wring more information out of the code.

Reply:
:+1: I can drop this for now, but I don't particularly like having these things that are hopefully fixed but possibly not

Reply:
@joelburget - @fosskers is testing things with Quickcheck to find out why this was failing.